src/main/java/net/oneandone/lavender/filter/processor/LavenderRewriteEngine.java

```diff
@@ -148,10 +148,14 @@ public URI rewrite(URI reference, URI baseURI, String contextPath) {
             return reference;
         }
 
-        return calculateURL(label, baseURI);
+        return calculateURL(label, baseURI, reference.getQuery());
     }
 
     public URI calculateURL(Label label, URI baseURI) {
+        return calculateURL(label, baseURI, null);
+    }
+
+    public URI calculateURL(Label label, URI baseURI, String query) {
```

Comment:
Could also be made "package private" (only used here and in the unit test). Then the previous method signature would also be obsolete.